src/Std/Data/DTreeMap/Internal/Lemmas.lean

```diff
@@ -1665,4 +1673,1216 @@ end Const
 
 end monadic
 
+theorem insertMany_cons (h : t.WF) {l : List ((a : α) × β a)} {k : α} {v : β k} :
+    (t.insertMany (⟨k, v⟩ :: l) h.balanced).1 =
+      ((t.insert k v h.balanced).impl.insertMany l h.insert.balanced).1 := by
+  simp only [insertMany, Id.run, Id.pure_eq, Id.bind_eq, List.forIn_yield_eq_foldl, List.foldl_cons]
+  rw [← List.foldl_hom Subtype.val, ← List.foldl_hom Subtype.val]
+  · exact fun t e => t.insert! e.1 e.2
+  · simp [insert_eq_insert!]
+  · simp [insert_eq_insert!]
```

Comment:
This seems to be quite a bit harder than for the hash map, do we understand why?

Reply:
Good point, I found a shorter proof. We can just use `simp [insertMany_eq_foldl, insert_eq_insert!]`.